src/main/java/de/srendi/advancedperipherals/common/addons/refinedstorage/RsApi.java

```diff
@@ -0,0 +1,679 @@
+     *
+     * @param resource the resource
+     * @param count    count of the resource - can be 0 and lower
+     * @return the parsed key to a lua properties map
+     */
+    public static Map<String, Object> getObjectFromResourceKey(@NotNull ResourceKey resource, long count) {
+        boolean countZeroOrLower = count <= 0;
+        if (resource instanceof ItemResource) {
+            if (countZeroOrLower) {
+                return getObjectFromItemResource(new ResourceAmount(resource, 1), count);
+            }
+            return getObjectFromItemResource(new ResourceAmount(resource, count));
+        }
+        if (resource instanceof FluidResource) {
+            if (countZeroOrLower) {
+                return getObjectFromFluidResource(new ResourceAmount(resource, 1), count);
+            }
+            return getObjectFromFluidResource(new ResourceAmount(resource, count));
+        }
+        AdvancedPeripherals.debug("Could not create table from unknown resource " + resource.getClass() + " - Report this to the maintainer of ap", Level.WARN);
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Helper method to get a parsed lua resourceAmount key. Currently only suppors item and fluid resources
+     *
+     * @param resourceAmount the resourceAmount
+     * @return the parsed key to a lua properties map
+     */
+    public static Map<String, Object> getObjectFromResourceAmount(@NotNull ResourceAmount resourceAmount) {
+        if (resourceAmount.resource() instanceof ItemResource) {
+            return getObjectFromItemResource(resourceAmount);
+        }
+        if (resourceAmount.resource() instanceof FluidResource) {
+            return getObjectFromFluidResource(resourceAmount);
+        }
+        AdvancedPeripherals.debug("Could not create table from unknown resourceAmount " + resourceAmount.getClass() + " - Report this to the maintainer of ap", Level.WARN);
+        return Collections.emptyMap();
+    }
+
+    public static Object parseDiskDrive(StorageNetworkNode diskDrive, InWorldNetworkNodeContainer nodeContainer) {
+        Map<String, Object> properties = new HashMap<>();
+
+        StorageConfiguration storageConfiguration = diskDrive.getStorageConfiguration();
+        CompositeStorage storage = (CompositeStorage) diskDrive.getStorage();
+
+        Set<Object> disks = new HashSet<>();
+        for (Storage disk : storage.getSources()) {
+            if (!(disk instanceof StateTrackedStorage stateTrackedStorage))
+                continue;
+
+            disks.add(parseStorageDisk(stateTrackedStorage));
+        }
+
+        properties.put("used", diskDrive.getStored());
+        properties.put("total", diskDrive.getCapacity());
+        properties.put("disks", disks);
+        properties.put("mode", storageConfiguration.getFilterMode().toString());
+        //TODO Redstone mode
+        properties.put("redstone_mode", false);
```

Comment:
If its a todo thing then I recommend we don't add this field right now? so people don't get confused when they saw a constant `false`

Reply:
ahh yeah, it is also not possible via rs to get that sate